go/vt/vttablet/tabletmanager/state_change.go

```diff
@@ -218,6 +218,18 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 					disallowQueryReason = "master tablet with filtered replication on"
 				}
 			}
+			if newTablet.Type == topodatapb.TabletType_REPLICA {
```

Comment:
is there a case where we would want something to serve if it has lag > unhealthy state? Possibly this should always if it's not a master

Reply:
good point. I was thinking at a minimum to also include RDONLY but "not master" makes more sense.